Add sprayer mission item

peterbarker opened the pull request.

rmackay9 commented on libraries/GCS_MAVLink/GCS_Common.cpp:
```diff
@@ -3710,6 +3711,20 @@ MAV_RESULT GCS_MAVLINK::handle_command_do_gripper(const mavlink_command_long_t &
+        return MAV_RESULT_FAILED;
+    }
+
+    sprayer->run(uint8_t(packet.param1));
```
run takes a boolean.. I'm sure this will work but I guess I would make it "is_positive(packet.param1)"

peterbarker replied:
I might just cast this to a boolean instead.  I'd like to keep non-boolean values Reserved For Future Use, if that's OK.

peterbarker replied:
I've adjusted this and the mission equivalent to strictly take 0 or 1 to allow for future modifications to the use of this value.

rmackay9 commented:
I wonder if it's good to keep putting mav-cmd into ardupilotmega.xml.  Might as well put them into common no?

rmackay9 commented:
This looks pretty good to me, thanks!

auturgy commented:
Agree, but if we want them in common they need to go via mavlink/mavlink else qgc will never get them. We really need to keep common sync’d.

> On 4 Jul 2020, at 9:54 am, Randy Mackay <notifications@github.com> wrote:
> 
> ﻿
> I wonder if it's good to keep putting mav-cmd into ardupilotmega.xml. Might as well put them into common no?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

amilcarlucas commented:
This can now go in

peterbarker commented:
@amilcarlucas I've rebased this and addressed @rmackay9 's comments.  I think this will need to have another dev merge it as it does update the mavlink submodule.

rmackay9 commented:
Looks good to me, thanks!  Feel free to merge after it passes CI.

peterbarker commented:
Can be merged after CI passes (careful with mavlink commit!)

amilcarlucas commented:
You missed the ArduPilot/mavlink#139 PR

CemalettinAkdogan commented:
Hi, I want to use command MAV_CMD_DO_SPRAY my agriculture drone in dronekit programming but I don't know how to run the command, Can you help me ?

amilcarlucas commented:
Read the dronekit documentation on how to send custom mavlink messages.
